### DISCUSSION TITLE
Macトラックパッドのピンチイン・ピンチアウトの再現方法はありますでしょうか？

### DISCUSSION THREAD

**Discussion by mura-** [Yowkees/keyball#106] [Q&A] [answered]
使用しているソフトウェアによってトラックパッドのピンチイン・ピンチアウトを使いたいことがあり、
トラックボールを使って再現する方法はありますでしょうか。
もしくはキーボードショートカットの拡大（CMD+）や縮小（CMD-）くらいしかできないのでしょうか。

**Comment by Yowkees** [comment, marked as answer]
現状のファームウェアではサポートしていなので、ソースファイルを修正する必要があります。
RemapでCMD+、CMD-を１キーの動作に登録できるので、それでも使い勝手は良さそうですが、
トラックボールを使用するとなるとやはりソースを変更してみるしかないかと。

**Reply by mura-** [reply]
なるほど、そうですよね、ありがとうございました。

**Comment by koron** [comment]
技術的な話になりますが
トラックボールの回転量に合わせたアナログ的なピンチイン・ピンチアウトを送る
のは実質不可能だと言えるでしょう。

根拠として
そもそもアナログでピンチイン及びピンチアウトのレポートをUSBを通じて送る方法が公開されていません。
(少なくとも私は見つけられていません)
もちろんQMKの機能としても実装されていません。

そのため仮にやるのであれば

1. ピンチイン・アウトをUSB HIDを通じて送信する仕様書を探し出し
2. QMK上に必要な実装を追加し
3. Keyballにそれを使うような実装を追加する

という3つのハードルを越える必要があります。
さすがにコレは無理があるなと考えています。

**Reply by mura-** [reply]
> そもそもアナログでピンチイン及びピンチアウトのレポートをUSBを通じて送る方法が公開されていません。

まさにそうですよね。。。
ピンチイン・ピンチアウト前提なアプリ（例えばFigmaなどのデザインツールの拡大・縮小など）が多数あり、
みなさんどうされてるのかなぁと思いきいてみました。

CMD+/CMD-などでみなさまやられてる、という認識にいたりました。

**Reply by naoto-tomita** [reply]
ちなみにFigmaは、レイヤー3でトラックボールを動かす（縦スクロールに固定されている状態）際に、Win(Command)を押しながらやると、ピンチイン・ピンチアウトできました！（エクセルではできなかったのでFigma固有のショートカットの様です）

**Reply by naoto-tomita** [reply]
@mura-

**Reply by mura-** [reply]
@naoto-tomita あれ、自分はそれうごかないですね。。。なんでだろう😂

**Reply by mura-** [reply]
すいません。できました。
Commandおしてから、Layer3+トラックボールでした💦

**Comment by mura-** [comment]
2年前の自己レスですが、
https://macmousefix.com/
これを使うことでピンチイン・ピンチアウトの再現はできました
